test/com/google/cose/EncryptionKeyTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cose;
+
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.Map;
+import co.nstant.in.cbor.model.NegativeInteger;
+import co.nstant.in.cbor.model.UnsignedInteger;
+import com.google.cose.exceptions.CoseException;
+import com.google.cose.utils.Headers;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Test class for testing {@link EncryptionKey}
+ */
+@RunWith(JUnit4.class)
+public class EncryptionKeyTest {
+  @Test
+  public void testRoundTrip() throws CborException, CoseException {
+    final String kVal = "65eda5a12577c2bae829437fe338701a10aaa375e1bb5b5de108de439c08551d";
+    final byte[] k = TestUtilities.hexStringToByteArray(kVal);
+    final String keyId = "meriadoc.brandybuck@buckland.example";
+    final Map map = new Map();
+    map.put(new UnsignedInteger(Headers.KEY_PARAMETER_KEY_TYPE),
+        new UnsignedInteger(Headers.KEY_TYPE_SYMMETRIC));
+    map.put(new UnsignedInteger(Headers.KEY_PARAMETER_KEY_ID), new ByteString(keyId.getBytes()));
+    map.put(new NegativeInteger(Headers.KEY_PARAMETER_K), new ByteString(k));
+
+    final EncryptionKey key = new EncryptionKey(map);
+    byte[] a = key.serialize();
+
+    final EncryptionKey newKey = EncryptionKey.parse(a);
+
+    Assert.assertEquals(Headers.KEY_TYPE_SYMMETRIC, newKey.getKeyType());
+    Assert.assertEquals(keyId, newKey.getKeyId());
+    Assert.assertEquals(new ByteString(k), newKey.getLabels().get(Headers.KEY_PARAMETER_K));
+
+    byte[] b = newKey.serialize();
+    Assert.assertEquals(a.length, b.length);
+    for (int i = 0; i < a.length; i++) {
+      Assert.assertEquals(a[i], b[i]);
```

Comment:
I'm curious how this passes with `Set` being used for operations now. Maybe we're just lucky? :)

Reply:
moved operations to a set instead of an arraylist. However, duplicate withOperations call can still happen and hence we cannot ensure only specific operations here.

Reply:
Set is only being used in builder class. parsing skips over the builder entirely and uses the encoded cbor key to generate the cose key.

Hence if Builder is not being used when rebuilding the entire key, we should be okay

Reply:
I think it works because we're using LinkedHashSet to hold the headers.

Reply:
Parse actually parses a CBOR Map, so it should be order preserving. We're fine :)